redux-form/lib/reduxForm.d.ts

```diff
@@ -285,65 +268,65 @@ export class SubmissionError<FormData extends DataShape> extends Error {
  * The following are methods or properties that you can access on an instance
  * of your decorated form component (outermost component named "ReduxForm").
  */
-export interface Form<FormData extends DataShape, P, S> extends Component<P, any> {
+export interface FormComponent<FormData extends DataShape, P, S> extends ComponentClass<P> {
```

Comment:
Naming this `Form` in the first place was not the best idea. Right now the library exports component named `Form` so the name is in conflict. In this case there is no way to avoid that unfortunately.